pkg/roll/run_update.go

```diff
@@ -238,10 +230,16 @@ func (u update) lockRCs() error {
 	}
 
 	err = u.lock.Lock(oldPath)
+	if err != nil {
+		// The second key couldn't be locked, so release the first key before retrying.
+		RetryOrQuit(
+			func() error { return u.lock.Unlock(newPath) },
+			done,
+			u.logger,
+			fmt.Sprintf("unlocking %s", newPath),
+		)
+	}
 	if _, ok := err.(kp.AlreadyLockedError); ok {
```

Comment:
I'm mostly concerned that there are cases where the `RetryOrQuit(unlock)` will go on forever. Particularly, if `lock.Lock` fails to acquire a lock, we get an `err`, and we enter this loop wherein `Unlock` will return an error every time we try to unlock our nonexistent lock (perhaps because our session expired between lock(newPath) and lock(oldPath)).